client/web/src/enterprise/batches/settings/CodeHostConnections.tsx

```diff
@@ -79,22 +79,37 @@ const CodeHostConnections: React.FunctionComponent<React.PropsWithChildren<CodeH
     const setupError = new URLSearchParams(location.search).get('error')
     const gitHubAppKindFromUrl = new URLSearchParams(location.search).get('kind')
     const shouldShowError = !success && setupError && gitHubAppKind !== GitHubAppKind.COMMIT_SIGNING
+    const gitHubAppInstallationInProgress =
+        success &&
+        gitHubAppKindFromUrl !== GitHubAppKind.COMMIT_SIGNING &&
+        (connection?.nodes.filter(n => n.credential).length ?? 0) === 0
     return (
         <Container className="mb-3">
             <H3>Code host credentials</H3>
             {headerLine}
             <ConnectionContainer className="mb-3">
                 {error && <ConnectionError errors={[error.message]} />}
                 {loading && !connection && <ConnectionLoading />}
-                {success && gitHubAppKindFromUrl !== GitHubAppKind.COMMIT_SIGNING && (
-                    <DismissibleAlert
-                        className="mb-3"
-                        variant="success"
-                        partialStorageKey={`batch-changes-github-app-integration-success-${appName}`}
-                    >
-                        GitHub App {appName?.length ? `"${appName}" ` : ''} successfully connected.
-                    </DismissibleAlert>
-                )}
+                {success &&
+                    gitHubAppKindFromUrl !== GitHubAppKind.COMMIT_SIGNING &&
+                    (gitHubAppInstallationInProgress ? (
+                        <DismissibleAlert
+                            className="mb-3"
+                            variant="info"
+                            partialStorageKey={`batch-changes-github-app-integration-pending-${appName}`}
+                        >
+                            GitHub App {appName?.length ? `"${appName}" ` : ''} is taking a few seconds to connect.
+                            Please refresh the page until the GitHub app appears.
```

Comment:
Requiring a manual refresh instead of doing some polling and refreshing in the background is fine for now (create a followup PR to add a more friendly polling/auto-refresh approach); can we make the refresh instructiuons more noticeable? I added a line break and bolding, in pure 1990s style as an example.

```suggestion
                            GitHub App {appName?.length ? `"${appName}" ` : ''} is taking a few seconds to connect.
                            <br/><b>Please refresh the page until the GitHub app appears.</b>
```

Reply:
Sounds good to me! Thank you for the suggestion :)

Reply:
I added SRCH-798 to keep track of the polling/status changes